conf/modules.config

```
@@ -1,32 +1,332 @@
+                path: { "${params.outdir}/samtools/" },
+                mode: params.publish_dir_mode
+            ],
+            [
+                pattern: "*.{flagstat,idxstats}",
+                path: { "${params.outdir}/samtools/" },
+                mode: params.publish_dir_mode,
+                enabled: params.save_samtools,
+                saveAs: { filename -> "${params.assembly ? 'user_assembly' : params.assembler}.${filename}" }
+            ]
+        ]
+    }
+
+    withName: PRODIGAL {
+        ext.args = { params.prodigal_trainingfile ? "-t $params.prodigal_trainingfile" : "-p meta" }
+        publishDir = [
+            path: { "${params.outdir}/prodigal" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: FORMAT_PRODIGAL_GFF {
+        publishDir = [
+            path: { "${params.outdir}/prodigal" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: PREDICT {
+        publishDir = [
+            path: { "${params.outdir}/transdecoder" },
+            mode: params.publish_dir_mode,
+            pattern: "*.transdecoder.*"
+        ]
+    }
+
+    withName: 'PROKKA' {
+        ext.args = '--prodigal --metagenome'
+    }
+
+    withName: 'FAA_CAT' {
+        ext.prefix = 'prokka.faa.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'FFN_CAT' {
+        ext.prefix = 'prokka.ffn.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'GFF_CAT' {
+        ext.prefix = 'prokka.gff.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: PROKKAGFF2TSV {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.prokka-annotations.tsv.gz"
+        ]
+    }
+
+    withName: '.*:FEATURECOUNTS_CDS' {
+        ext.args = '-g ID -t CDS -F gtf'
+        publishDir = [
+            path: { "${params.outdir}/featurecounts" },
+            mode: params.publish_dir_mode,
+            pattern: "*.featureCounts.*"
+        ]
+    }
+
+    withName: COLLECT_FEATURECOUNTS {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'HMMRANK' {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.tsv.gz"
+        ]
+    }
+
+    withName: EGGNOG_DOWNLOAD {
+        storeDir = "${params.eggnog_dbpath}"
```

```suggestion
        storeDir = { "${params.eggnog_dbpath}" }
```
Does `storeDir` accept a closure here, or does it accept strings only?

---

I think so. At least when I tried without the closure it didn't work!